test/e2e/e2e_test.go

```diff
@@ -335,21 +354,26 @@ func createCluster(ctx context.Context, client *godo.Client, nameSuffix, kubeMaj
 		page = p + 1
 	}
 
+	if deleted {
+		// Sleeping briefly to reach consistent state.
```

Comment:
I am assuming by "reach consistent state" you mean "don't try to create before the old cluster has finished deleting"; what do you think about a wait loop that waits until a `client.Kubernetes.Get` (or whatever the equivalent method is called) returns a 404?

Probably not a huge deal since we don't run this suite very often, but we do do something like that in our internal test runner suite.

Reply:
That makes a lot of sense to me. I implemented your suggestion and updated the PR accordingly.

While I was doing so, I realized why the PR would fail frequently. Initially I blamed it on Github [Actions], but then it turned out that the extra snapshot-only E2E test run I added was using the same cluster name as the 1.17 run. Consequently, they were racing for the same cluster, which is why one might see a 404 right after the cluster create when it overlapped with the other run deleting what it thought to be a left-over cluster from a previous run.
I made a small modification to the workflow to add another suffix to the cluster name when we run the snapshot-only tests. It's not pretty but it gets the job done until we can (hopefully soon) use the full 1.17 E2E suite from upstream.